### PR TITLE
hope

### PR DESCRIPTION
**Pull request by julia951**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,53 @@
-# yet_another_tests
\ No newline at end of file
+# Yet Another Tests
+
+This project contains automated tests using Playwright and Mocha for the Sauce Demo website.
```
Comment (ovartem) [inline README.md]: mocha is not used with playwright

File: README.md [hunk 2]
```diff
@@ -1 +1,53 @@
...
+
+## Project Structure
+
+- `tests/`: Contains the test files.
+  - `firstTest.js`: Test for performing login and verifying elements on the page.
+  - `secondTest.js`: Test for adding a product to the cart and verifying the cart contents.
+- `package.json`: Contains the project dependencies and scripts.
+- `mocha.opts`: Configuration file for Mocha.
```
Comment (ovartem) [inline README.md]: mocha

File: README.md [hunk 3]
```diff
@@ -1 +1,53 @@
...
+- `README.md`: This file.
+
+## Setup
+
+### 1. Install Dependencies
+
+Make sure you have Node.js installed. Then, install the necessary dependencies:
+
+npm install
```
Comment (ovartem) [inline README.md]:
```suggestion
`npm install`
```

File: mocha.opts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+--recursive
```
Comment (ovartem) [inline mocha.opts]: this file can be removed

File: package.json [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    "scripts": {
+      "test": "mocha tests",
+      "test-login": "mocha tests/firstTest.js",
+      "test-cart": "mocha tests/secondTest.js"
+    },
+    "author": "",
+    "license": "ISC",
+    "dependencies": {
+      "playwright": "^1.20.0",
+      "mocha": "^9.1.3",
```
Comment (ovartem) [inline package.json]: you can delete this record and run `npm install`

File: package.json [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+      "assert": "^2.0.0"
```
Comment (ovartem) [inline package.json]: this lib is not needed

File: tests/firstTest.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+  await page.fill('#password', 'secret_sauce');
+  await page.click('#login-button');
+
+  const productTitle = await page.isVisible('.title');
+  console.log(`Products title is displayed: ${productTitle}`);
+
+  const shoppingCartIcon = await page.isVisible('.shopping_cart_link');
+  console.log(`Shopping Cart icon is displayed: ${shoppingCartIcon}`);
+
+  const productCount = await page.$$eval('.inventory_item', items => items.length);
```
Comment (ovartem) [inline tests/firstTest.js]:
```suggestion
  const productCount = await page.locator('.inventory_item').count();
```

Comment (ovartem) [reply tests/firstTest.js]: https://playwright.dev/docs/api/class-locatorassertions#locator-assertions-to-have-count

File: tests/firstTest.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
+const { chromium } = require('playwright');
```
Comment (ovartem) [inline tests/firstTest.js]: this is not a test

File: tests/secondTest.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
+const { chromium } = require('playwright');
+const assert = require('assert');
+
+describe('Add product to the cart', function () {
```
Comment (ovartem) [inline tests/secondTest.js]: you can use test.describe instead

File: tests/secondTest.js [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+  });
+
+  it('should login, add product to cart, and verify', async () => {
+    await page.goto('https://www.saucedemo.com/');
+
+    await page.fill('#user-name', 'standard_user');
+    await page.fill('#password', 'secret_sauce');
+    await page.click('#login-button');
+
+    await page.click('.inventory_item button:has-text("Add to cart")');
```
Comment (ovartem) [inline tests/secondTest.js]: this is not a generic selector

 should be changed to more generic

**Review by ovartem — APPROVED** [review]
Дуже добре! не без нюансів, але початок дуже хороший, зараз давай не зупинятись. Скину тобі ще відео пояснення

File: test-results/.last-run.json [hunk 11]
```diff
@@ -0,0 +1,4 @@
+{
+  "status": "failed",
+  "failedTests": []
+}
```
Comment (AlinaZhykharieva) [inline test-results/.last-run.json:4]: Hi test results also should bé added to git ignore

File: node_modules/.package-lock.json [hunk 12]
```diff
@@ -0,0 +1,50 @@
+{
```
Comment (hlazkov) [inline node_modules/.package-lock.json:1]: we dont need this file.

File: test-results/.last-run.json (hunk 11 above)
Comment (hlazkov) [reply test-results/.last-run.json:4]: For sure!

File: tests/secondTest.spec.js [hunk 13]
```diff
@@ -0,0 +1,14 @@
+const { test, expect } = require('@playwright/test');
+
+test('Sauce Demo product display test', async ({ page }) => {
+  await page.goto('https://www.saucedemo.com/');
```
Comment (hlazkov) [inline tests/secondTest.spec.js]: lets store this link in baseUrl inside of playwright.config

**Review by hlazkov — APPROVED** [review]
Seems pretty good. It would be great to resolve a few issues before merging it. Nothing critical, but I would recommend to pay more attention to gitignore. Comment regarding baseUrl is JFYI

File: .gitignore  [hunk 14]
```diff
@@ -0,0 +1,5 @@
+node_modules/
+.node_modules/
+node_modules
+package-lock.json
```
Comment (hlazkov) [inline .gitignore :4]: we need `package-lock.json` but we dont need `node_modules/package-lock.json`

File: .gitignore  [hunk 15]
```diff
@@ -0,0 +1,5 @@
...
+test-results
```
Comment (hlazkov) [inline .gitignore :5]:
```suggestion
test-results/
```

File: .gitignore  (hunk 14 above)
Comment (hlazkov) [reply .gitignore :4]: You need to just remove it from staged files same with test report

**Review by hlazkov — APPROVED** [review]
You could clean up this PR or proceed to the ned unit as is.

**Review by AlinaZhykharieva — APPROVED** [review]

**Pull request merged**